surfactant/relationships/dotnet_relationship.py

```diff
@@ -65,6 +66,58 @@ def establish_relationships(
                                 dnProbingPaths = []
                             dnProbingPaths.append(pathlib.PureWindowsPath(path).as_posix())
 
+    # https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/loading-unmanaged
+    # 1. Check the active AssemblyLoadContext cache
+    # 2. Calling the import resolver set by the setDllImportResolver function
+    #    - a. Example using SetDllImportResolver: https://learn.microsoft.com/en-us/dotnet/standard/native-interop/native-library-loading
+    #    - b. Checks PInvoke's or Assembly's DefaultDllImportSearchPathsAttribute, then the assembly's directory, then LoadLibraryEx with LOAD_WITH_ALTERED_SEARCH_PATH flag
+    #       - This attribute has no effect on non-Windows platforms / Mono runtime
+    #       - https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.defaultdllimportsearchpathsattribute?view=net-7.0
+    #       - i. This has a "Paths" property which is a bitwise combination of paths specified in ii:
+    #       - ii. https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.dllimportsearchpath?view=net-7.0
+    # 3. The active AssemblyLoadContext calls its LoadUnmanagedDll function (Default behavior is the same as AssemblyRef probing?)
+    #    - a. Can be overridden, but the default implementation returns IntPtr.Zero, which tells the runtime to load with its default policy.
+    #    - b. https://learn.microsoft.com/en-us/dotnet/api/system.runtime.loader.assemblyloadcontext.loadunmanageddll?view=net-7.0
+    # 4. Run default unmanaged library probing logic by parsing *.deps.json probing properties
+    #    - a. If the json file isn't present, assume the calling assembly's directory contains the library
+    #    - b. https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/default-probing#unmanaged-native-library-probing
+    if "dotnetImplMap" in metadata:
+        for asmRef in metadata["dotnetImplMap"]:
+            if "Name" not in asmRef:
+                continue
+            refName = asmRef["Name"]
+            if is_absolute_path(refName):
+                relationships.extend(get_windows_pe_dependencies(sbom, software, [refName]))
+                continue
+            probedirs = []
+            if isinstance(software.installPath, Iterable):
+                for ipath in software.installPath:
+                    probedirs.append(pathlib.PureWindowsPath(ipath).parent.as_posix())
+            combinations = [refName]
+            if not (refName.endswith(".dll") or refName.endswith(".exe")):
+                combinations.append(f"{refName}.dll")
+            combinations.extend(
+                [
+                    f"{refName}.so",
+                    f"{refName}.dylib",
+                    f"lib{refName}.so",
+                    f"lib{refName}.dylib",
+                    f"lib{refName}",
+                ]
+            )
```

Comment:
I'm not familiar with the actual behavior for how .NET will probe for files here, and the online doc page doesn't seem to touch on this -- if refName is `abc.dll`, then on Linux will it search for a file named `abc.dll.so`, or does .NET try to be smart and remove the `.dll` so before appending `.so`?

Reply:
Issue #80 - backlog for something to check eventually